File: LempelZivAlgorithm/LempelZivAlgorithm/Decompressor.cpp
#include "Decompressor.h"
#include <vector>
#include <iostream>
#include "DecompressorHelpers.h"
using namespace std;

Decompressor::Decompressor()
{
}


Decompressor::~Decompressor()
{
}

void Decompressor::Decompress(vector<bool> &encodedData, vector<char> &decodedData)
{
	vector< vector<char> > dictionary = InitializeDictionary(); //Initialize dictionary
	vector<char> outputSequence;
	vector<char> entry;
	int decimalCounter = 255;
	int temp = 0;
	outputSequence = Read(encodedData, temp, dictionary, decimalCounter++);
	decodedData.insert(decodedData.end(), outputSequence.begin(), outputSequence.end());
	entry.insert(entry.end(), outputSequence.begin(), outputSequence.end());
	int aaaaa = 5;

	for(int i=8;i<encodedData.size();)
	{
		outputSequence = Read(encodedData, i, dictionary, decimalCounter);
		if(outputSequence[0] == 'O')
		{
			int aaa = 5;
		}
		decodedData.insert(decodedData.end(), outputSequence.begin(), outputSequence.end());
		entry.insert(entry.end(), outputSequence.begin(), outputSequence.begin()+1);
		dictionary.push_back(entry);
		decimalCounter++;
		entry.clear();
		entry.insert(entry.end(), outputSequence.begin(), outputSequence.end());
	}
	std::cout << aaaaa;

}

File: LZWAlgorithm/LZWAlgorithm/Decompressor.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LZWAlgorithm
{
    class Decompressor
    {
        /// <summary>
        /// ByteGetter yra sukurta klase, kuri reikalinga tam, kad butu
        /// patogiau nuskaitomi List byte pagal encodedData 
        /// </summary>
        /// <param name="inputFileName"></param>
        /// <param name="outputFileName"></param>
        public static void Decompress(string inputFileName, string outputFileName)
        {
            List<byte> decodedData = new List<byte>();
            byte[] encodedDataArray = File.ReadAllBytes(inputFileName);
            List<List<byte>> dictionary = InitializeDictionary();
            ByteGetter bg = new ByteGetter(encodedDataArray.ToList(), dictionary);

            List<byte> w = bg.NextByteList();

            while (!bg.IsEmpty())
            {
                var output = bg.NextByteList();
                decodedData.AddRange(output);//adds output to decodedData
                w.Add(output[0]); //adds first byte from output to w, which is then added to dictionary
                dictionary.Add(w);//adds new entry
                w.RemoveAt(w.Count - 1); //removes last element to leave only the conjecture
            }

            File.WriteAllBytes(outputFileName, decodedData.ToArray());

        }

        public static List<List<byte>> InitializeDictionary()
        {
            List<List<byte>> result = new List<List<byte>>();

            for (int i = 0; i < 256; i++)
            {
                List<byte> temp = new List<byte> {(byte) i};
                result.Add(temp);
            }

            return result;
        }
    }
}
